### ISSUE TITLE
GameGuru MAX - Request - Some Kind of Protection for Stand Alone Games?

### ISSUE THREAD

**Issue by ArgentArts** [TheGameCreators/GameGuruRepo#2442]
After looking in a stand alone game folder, it's plane to see that if anyone purchases a game made with GameGuru MAX and also has GameGuru MAX, they can access the entire game using their copy of MAX. Every file is exposed and easy to open since everything is there in formats that MAX opens and reads (.fpm, .dbo, etc.). All of our models are easily accessible in the files>entitybank>user folder where someone could just copy them over to their own user folder and ... well ... use them.

I would like to request a means to add a level of protection to our games, even if it's a simple one, like a "package" that will hold all the files (like ZIP does) or that the stand alone spits out file formats that only it can read (even if it just changes the last names so that the average person doesn't know what these files are just by looking at them.

**Comment by fearlesswee** [comment]
This is extremely important, especially considering the terms-of-service of many asset stores have rules against distributing unprotected copies of their assets, as this is not really that different than buying an asset and intentionally distributing it for free. Could land users in some legal hot water without them even being aware of it! (Not to mention users who make their own custom assets for their game don't want to be handing them out for free either!)

**Comment by ArgentArts** [comment]
Exactly. Let's say I purchase a commercial license to use a model in my game. It gives me the right to use this model in my game. But because MAX does not provide any means of protecting my game once I create a standalone and start to sell or distribute my game on Steam or elsewhere, now everyone has free access to this commercial asset and I am liable because I've distributed it.

Fortunately, I make all my own assets. Unfortunately, I make all my own assets ... meaning ... I don't want anyone to just run off and use them without my permission.

**Comment by synchromesh62** [comment]
At the moment we are not allowed to Sell or Publish games until after EA so we are going to have to be choosy on who we give them to unless you do not mind your assets being being used by other users and not forgetting it includes custom scripts, textures and audio as well. Course you could add your own licence to your game but probably easier not to spread your stuff until its protected.

**Comment by ArgentArts** [comment]
That's not the issue and didn't need to be said. I am not planning on selling a game made with MAX any time soon. It may be a year or two before I have anything even remotely ready. The issue is there is nothing currently in the works to protect our assets in the eventuality that someone does make a game they want to sell. Therefore, I've made a **request** (as the title says) so that TGC will consider this so that we may have it at some point in the future. That's part of what GitHub is for, after all.

**Comment by synchromesh62** [comment]
**The issue is, there is nothing currently in the works to protect our assets in the eventuality that someone does make a game they want to sell at any point in the future (once we are cleared to do so).**

It has a **Post Early Access** label meaning it will be done some time after the EA release  ?
The encryption will probably be the same as classic so no doubt its ready to go in whenever.
I was agreeing with you and saying for now we would have to be weary of who we give our demos to etc.

**Comment by ArgentArts** [comment]
It has a Post Early Access label because that is the label they gave my request after I made the request. ;)

I was just pointing out that it was not necessary to tell us that we were not allowed to sell or publish pre-release games. The splash screen tells us this each time we start it up. ;) It just seemed a bit rude, though you probably didn't intend it that way. Sorry that I took it that way. I apologize. I've actually not been feel very well at all today. It shouldn't be an excuse, but my head is killing me and I think it's affecting my mood more than it should.

**Comment by synchromesh62** [comment]
Apologies it was not meant to be rude 
I was responding to this one you posted 4 hours ago 
**" I create a standalone and start to sell or distribute my game on Steam or elsewhere, now everyone has free access to this commercial asset and I am liable because I've distributed it. "**
I just meant we could not do that anyway until EA is over and by that time encryption will be in. 
I think we just got crossed wires somewhere so apologies again. I just thought it was an innocent comment in response.
